homeassistant/helpers/intent.py

```diff
@@ -117,6 +124,72 @@ def __repr__(self):
         return '<{} - {}>'.format(self.__class__.__name__, self.intent_type)
 
 
+def fuzzymatch(name, entities):
+    """Fuzzy matching function."""
+    matches = []
+    pattern = '.*?'.join(name)
+    regex = re.compile(pattern)
+    for entity in entities:
+        match = regex.search(entity)
+        if match:
+            matches.append((len(match.group()), match.start(), entity))
+    return [x for _, _, x in sorted(matches)]
+
+
+class ServiceIntentHandler(IntentHandler):
+    """Service Intent handler registration.
+
+    Service specific intent handler that calls a service by name/entity_id.
+    """
+
+    slot_schema = {
+        'name': cv.string,
+    }
+
+    def __init__(self, intent_type, domain, service, speech):
+        """Create Service Intent Handler."""
+        self.intent_type = intent_type
+        self.domain = domain
+        self.service = service
+        self.speech = speech
+
+    @asyncio.coroutine
+    def async_handle(self, intent_obj):
+        """Handle the hass intent."""
+        hass = intent_obj.hass
+        slots = self.async_validate_slots(intent_obj.slots)
+        response = intent_obj.create_response()
+
+        name = slots['name']['value']
+        entities = {state.entity_id: state.name for state
+                    in hass.states.async_all()}
+        entity_name = name.replace(' ', '_').lower()
+        entity_name = entity_name.replace('the_', '')
```

Comment:
Please add a comment why this is? Also, it seems very English focused

Reply:
True, and again with the above this may no longer be necessary but then again it might, but this logic should be moved into that function. But a typical utterance is 'Turn on the living room lights' so we get 'the living room lights' for the entity name. Not that we are trying to match everything  but it seemed a common enough use. But I can remove it since there is no need to make it too specific and turn on livign room lights is fine.

Reply:
This sounds like an issue that should be fixed by things launching intents?

Reply:
Yep, makes sense that it would be done by extending utterances instead.